practicas/dianasalgado22/P1/01_niveles_linguisticos.py

```diff
@@ -0,0 +1,689 @@
+    return result
+
+
+def get_ipa_transcriptions(word: str, dataset: dict) -> list[str]:
+    """Search for a word in an IPA phonetics dict
+
+    Given a word this function return the IPA transcriptions
+
+    Parameters:
+    -----------
+    word: str
+        A word to search in the dataset
+    dataset: dict
+        A dataset for a given language code
+
+    Returns
+    -------
+    list[str]:
+        List with posible transcriptions if any,
+        else an empty list
+    """
+    return dataset.get(word.lower(), "").split(", ")
+
+
+# #### Obteniendo todos los datos
+
+IPA_URL = "https://raw.githubusercontent.com/open-dict-data/ipa-dict/master/data/{lang}.txt"
+
+lang_codes = {
+    "ar": "Arabic (Modern Standard)",
+    "de": "German",
+    "en_UK": "English (Received Pronunciation)",
+    "en_US": "English (General American)",
+    "eo": "Esperanto",
+    "es_ES": "Spanish (Spain)",
+    "es_MX": "Spanish (Mexico)",
+    "fa": "Persian",
+    "fi": "Finnish",
+    "fr_FR": "French (France)",
+    "fr_QC": "French (Québec)",
+    "is": "Icelandic",
+    "ja": "Japanese",
+    "jam": "Jamaican Creole",
+    "km": "Khmer",
+    "ko": "Korean",
+    "ma": "Malay (Malaysian and Indonesian)",
+    "nb": "Norwegian Bokmål",
+    "nl": "Dutch",
+    "or": "Odia",
+    "ro": "Romanian",
+    "sv": "Swedish",
+    "sw": "Swahili",
+    "tts": "Isan",
+    "vi_C": "Vietnamese (Central)",
+    "vi_N": "Vietnamese (Northern)",
+    "vi_S": "Vietnamese (Southern)",
+    "yue": "Cantonese",
+    "zh_hans": "Mandarin (Simplified)",
+    "zh_hant": "Mandarin (Traditional)"
+}
+iso_lang_codes = list(lang_codes.keys())
+
+
+def get_corpora() -> dict:
+    """Download corpora from ipa-dict github
+
+    Given a list of iso lang codes download available datasets.
+
+    Returns
+    -------
+    dict
+        Lang codes as keys and dictionary with words-transcriptions
+        as values
+    """
+    return {
+        code: parse_response(download_ipa_corpus(code))
+         for code in iso_lang_codes
+        }
+
+
+data = get_corpora()
+
+# #### Sistema de búsqueda  
+
+# +
+from Levenshtein import distance as levenshtein_distance
+
+def buscar_palabras_similares(palabra, dicc , top_n=3):
+    """
+    Encuentra las n palabras más similares en el diccionario usando la distancia de Levenshtein.
+    """
+    palabras_ordenadas = sorted(dicc.keys(), key=lambda x: levenshtein_distance(palabra, x))[:top_n]
+    return palabras_ordenadas
+
+
+# -
+
+def get_formated_string(code: str, name: str) -> str:
+    """
+    Devuelve una cadena formateada con negritas y color amarillo para su uso con `rich.print`.
+
+    Args:
+        code (str): Código del idioma.
+        name (str): Nombre del idioma.
+
+    Returns:
+        str: String formateado con etiquetas de `rich`.
+    """
+    return f"[bold]{name.strip()}[/bold]\n[yellow]{code.strip()}"
+
+
+def process_query(query: str, sub_dataset: Dict[str, List[str]]) -> None:
```

Comment:
FYI: desde `python >= 3.9` puedes usar los tipos nativos de python

 ```suggestion
def process_query(query: str, sub_dataset: dict[str, list[str]]) -> None:
```